mlfinlab/tests/test_sb_bagging.py

```diff
@@ -0,0 +1,310 @@
+
+        sb_clf_1 = SequentiallyBootstrappedBaggingClassifier(base_estimator=clf_1, max_features=0.2,
+                                                             n_estimators=100,
+                                                             triple_barrier_events=self.meta_labeled_events,
+                                                             price_bars=self.data, oob_score=True,
+                                                             random_state=1, bootstrap_features=True,
+                                                             max_samples=30, verbose=2)
+
+        sb_clf_2 = SequentiallyBootstrappedBaggingClassifier(base_estimator=clf_2, max_features=7,
+                                                             n_estimators=100,
+                                                             triple_barrier_events=self.meta_labeled_events,
+                                                             price_bars=self.data, oob_score=False,
+                                                             random_state=1, bootstrap_features=True,
+                                                             max_samples=0.3, warm_start=True)
+
+        sb_clf_3 = SequentiallyBootstrappedBaggingClassifier(base_estimator=clf_3,
+                                                             triple_barrier_events=self.meta_labeled_events,
+                                                             price_bars=self.data)
+
+        sb_clf_4 = SequentiallyBootstrappedBaggingClassifier(base_estimator=clf_4, max_features=0.2,
+                                                             n_estimators=100,
+                                                             triple_barrier_events=self.meta_labeled_events,
+                                                             price_bars=self.data, oob_score=True,
+                                                             random_state=1, bootstrap_features=True,
+                                                             max_samples=30, verbose=2)
+
+        sb_clf_1.fit(self.X_train, self.y_train_clf, sample_weight=np.ones((self.X_train.shape[0],)), )
+
+        sb_clf_2.fit(self.X_train, self.y_train_clf, sample_weight=np.ones((self.X_train.shape[0],)), )
+        sb_clf_2.n_estimators += 0
+        sb_clf_2.fit(self.X_train, self.y_train_clf, sample_weight=np.ones((self.X_train.shape[0],)), )
+        sb_clf_2.n_estimators += 2
+        sb_clf_2.fit(self.X_train, self.y_train_clf, sample_weight=np.ones((self.X_train.shape[0],)), )
+
+        sb_clf_3.fit(self.X_train, self.y_train_clf)
+        sb_clf_4.fit(self.X_train, self.y_train_clf)
+
+    def test_value_error_raise(self):
+        """
+        Test various values error raise
+        """
+        clf = KNeighborsClassifier()
+        bagging_clf_1 = SequentiallyBootstrappedBaggingClassifier(base_estimator=clf,
+                                                                  triple_barrier_events=self.meta_labeled_events,
```

Comment:
here we pass through the meta label events when we instantiate. Would it not be better if this was rather part of the fit function? I am sure there is a reason it's added here. I am just thinking of the implications. Would it then matter which data I pass to the fit method, will there be a mismatch between the self.meta_labeled_events and the fit data?

Reply:
We need to pass to it constructor, it is needed during cross-validation. We generate indicator matrix using all meta events, and when we fit we subsample that ind_mat so that only sample from X are used